client/lib/viewport/react-helpers.js

```diff
@@ -0,0 +1,123 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect } from 'react';
+import { camelCase, upperFirst } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	isWithinBreakpoint,
+	subscribeIsWithinBreakpoint,
+	MOBILE_BREAKPOINT,
+	DESKTOP_BREAKPOINT,
+} from './index';
+
+/**
+ * React hook for getting the status for a breakpoint and keeping it updated.
+ *
+ * @param {String} breakpoint The breakpoint to consider.
+ *
+ * @returns {Boolean} The current status for the breakpoint.
+ */
+export function useBreakpoint( breakpoint ) {
+	const [ isActive, setIsActive ] = useState( () => isWithinBreakpoint( breakpoint ) );
+
+	function handleBreakpointChange( currentStatus ) {
+		setIsActive( currentStatus );
+	}
```

Comment:
This function could be defined inside the `useEffect` callback. Or even omitted in favor of passing `setIsActive` directly as the listener. It's guaranteed to be identical for a given component instance.

Reply:
Moved.